test/testdefs.jl

```diff
@@ -15,14 +15,26 @@ function runtests(name, path, isolate=true; seed=nothing)
             m = Main
         end
         @eval(m, using Test, Random)
+        @eval(m, import Pkg)
         let id = myid()
             wait(@spawnat 1 print_testworker_started(name, id))
         end
+        run_tests = if startswith(path, Sys.STDLIB)
+            env = dirname(dirname(path))
+            quote
+                Pkg.activate($env)
```

Comment:
We run many of the tests for a given stdlib in parallel but with this it seems it will run the full test suite for them over and over. Also, Pkg.test is pretty noisy and having a nu ch of these processes going St the same time will make the rest of the test logs hard to read.

Reply:
See e.g. https://github.com/JuliaLang/julia/blob/master/stdlib/LinearAlgebra/test/testgroups

Reply:
It might also be kinda racy when all the workers are fighting to download the registry and the test dependencies and put them in the same place.

Reply:
The goal here is to be able to test stdlibs that have non-stdlib test dependencies. I don't really care how that's accomplished, I'm pretty much just flailing around here trying to find something that works.